### PR TITLE
chore: add github action to publish to npm

### PR DESCRIPTION
**Pull request by roziscoding**
The action will run whenever a `v1.*` tag is created pointing to `main`.
It will call `npm publish`, which will run `npm run backport`
because of the `prepare` npm script.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -13,6 +13,13 @@ jobs:
             - name: Checkout
               uses: actions/checkout@v3
 
+            - name: Publish to npm
+              run: |
+                npm config set //registry.npmjs.org/:_authToken '${NPM_TOKEN}'
+                npm publish
+              env :
+                NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
+
```
Comment (roj1512) [inline .github/workflows/release.yml:22]: No build?

Comment (roziscoding) [reply .github/workflows/release.yml:22]: I forgot that publish doesn't run prepare.
I'll add the build step later today

Comment (roziscoding) [reply .github/workflows/release.yml:25]: Done

Comment (wojpawlik) [reply .github/workflows/release.yml:25]: It does, but https://snyk.io/blog/github-actions-to-securely-publish-npm-packages/

Also, `prepare` requires `npm ci --ignore-scripts`.

Comment (KnorpelSenf) [reply .github/workflows/release.yml:25]: At this point it would almost be simpler to just `npm install`

Comment (KnorpelSenf) [reply .github/workflows/release.yml:25]: But yeah … npm ci etc is preferable, I'm not advocating for npm install … forget what I said

Comment (roziscoding) [reply .github/workflows/release.yml:25]: Yeah, it explicity runs `backport` now

Comment (KnorpelSenf) [reply .github/workflows/release.yml:25]: @wojpawlik is still right that you'll need to install the deps first

Comment (roziscoding) [reply .github/workflows/release.yml:25]: Hm... true

Comment (roziscoding) [reply .github/workflows/release.yml:25]: Should be good now

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -3,7 +3,7 @@ name: Main
 on:
     push:
         tags:
-            - "*"
+            - "v1.*"
```
Comment (roj1512) [inline .github/workflows/release.yml:6]: Why not `"*"`?

Comment (KnorpelSenf) [reply .github/workflows/release.yml:6]: If something goes wrong, we don't accidentally want to publish major versions or other tags

**Review by KnorpelSenf — APPROVED** [review]

File: .github/workflows/release.yml (hunk 2 above)
Comment (roj1512) [reply .github/workflows/release.yml:6]: Cool.

**Review by roj1512 — APPROVED** [review]

**Comment by roziscoding** [comment]
Should we :shipit: ?

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -13,6 +13,16 @@ jobs:
             - name: Checkout
               uses: actions/checkout@v3
 
+            - name: Backport
+              run: npm ci --ignore-scripts && npm run backport
```
Comment (wojpawlik) [inline .github/workflows/release.yml:17]: I'd split this into 2 steps, to make it easier to read logs:
```suggestion
            - run: npm ci --ignore-scripts
            - run: npm run backport
```

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -13,6 +13,16 @@ jobs:
...
+
+            - name: Publish to npm
+              run: |
+                npm config set //registry.npmjs.org/:_authToken '${NPM_TOKEN}'
+                npm publish
```
Comment (wojpawlik) [inline .github/workflows/release.yml:22]:
```suggestion
                npm publish --ignore-scripts
```

File: .github/workflows/release.yml [hunk 5]
```diff
@@ -13,6 +13,16 @@ jobs:
...
+              env :
```
Comment (wojpawlik) [inline .github/workflows/release.yml:23]: Nitpick:
```suggestion
              env:
```

**Review by wojpawlik — APPROVED** [review]

**Pull request merged**